sklearn/_config.py

```diff
@@ -133,6 +135,17 @@ def set_config(
 
```

Comment:
Could you please update the `engine_provider` entry above to mention that it's possible to either register the name of an entry point or directly pass an engine class instance for testing purpose?

Or do we want to keep the engine class backdoor "secret" for now?

Reply:
Let's do this for now.

Since this API is explicitly experimental we can still change our mind later.